ocaml/xapi-idl/lib/xcp_service.ml

```diff
@@ -163,6 +163,25 @@ let setify =
   in
   loop []
 
+(** How long to let an OCaml thread run, before
+    switching to another thread.
+    This needs to be as small as possible to reduce latency.
+
+    Too small values reduce performance due to context switching overheads
+  
+    4ms = 1/HZ in Dom0 seems like a good default,
+    a better value will be written by a boot time service.
+ *)
+let timeslice = ref 0.05
+
+let apply_timeslice () =
+  let interval = !timeslice in
+  D.debug "Setting timeslice to %.3fs" interval ;
+  if interval >= 0.05 then
+    D.debug "Timeslice same as or larger than default: not setting"
```

Comment:
The where is `Xapi_timeslice.Timeslice.set` called for the default case?

Reply:
Oh you mean the OCaml default of 50 ms.

Reply:
Yeah the default timeslice is always there, this is in *addition to that*, it is not possible to turn off the default 50 ms switching.

Reply:
So the idea is to add additional yields; would suggest to add `__FUNCTION__` to log messages.